### DISCUSSION TITLE
What is the purpose of isSubmitting in form state?

### DISCUSSION THREAD

**Discussion by paschaldev** [TanStack/form#952] [Q&A] [answered]
The purpose of isSubmitting is not clear. How do you control this value, how do you update the state of `isSubmitting`, is this automated? When does it trigger?

I would like to set `isSubmitting` to true when I'm running an async request to store something on the server from the form values but I don't see how to use the `isSubmitting` in the docs to do this even though it's present everywhere in examples.

The DOC should be clear on the purpose of `isSubmitting` and how to use this value. I want to show a spinner when the form `isSubmitting` after wrapping the button with `form.subscribe`

**Comment by Balastrong** [comment, marked as answer]
It is indeed automatic and is true during the submit process, goes false when form is no longer submitting as the name suggests and is exactly for your use case.

Extra explanation: it is set to true when `form.handleSubmit` and is set to false when either:
A) Validation is performed and form is invalid
or
B) Form is valid, onSubmit is awaited and it fulfills

Here for more context (look where `done()` is called)
https://github.com/TanStack/form/blob/ade193c2bb68999bcb921c5aa899e4ef950f509a/packages/form-core/src/FormApi.ts#L913-L968


I'll try to make a guess, are you awaiting for your async call inside your onSubmit?

**Reply by MartinCura** [reply]
> I'll try to make a guess, are you awaiting for your async call inside your onSubmit?

You guessed right!... for someone else, 1 year in the future 😄 grazie!

**Reply by paschaldev** [reply]
Oh yes. Thank you
